sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/spice_indexer.py

```diff
@@ -0,0 +1,386 @@
+        (left, right) = spiceypy.wnfetd(cover, i_window)
+        results_j2000.append([left, right])
+        # 5) Convert the time span to datetime
+        results_datetime.append(
+            [spiceypy.et2datetime(left), spiceypy.et2datetime(right)]
+        )
+        # 6) Convert the time span to spacecraft clock time
+        results_sclk.append(
+            [
+                spiceypy.sce2s(SPACECRAFT_ID, left),
+                spiceypy.sce2s(SPACECRAFT_ID, right),
+            ]
+        )
+
+    return results_j2000, results_datetime, results_sclk
+
+
+def _upsert_into_spice_table(
+    spice_object: SPICEFilePath,
+    file_coverage_j2000: list[list[float]],
+    file_coverage_datetime: list[list[datetime]],
+    file_coverage_sclk: list[list[str]],
+    latest_sclk: Path,
+    latest_lsk: Path,
+):
+    """Insert/Update the spice metadata table with collected data.
+
+    Parameters
+    ----------
+    spice_object: SPICEFilePath
+        The SPICE file to upsert
+    file_coverage_j2000: list[list[float]]
+        A list of file intervals in j2000 time format
+    file_coverage_datetime: list[list[datetime]]
+        A list of file intervals in datetime format
+    file_coverage_sclk: list[list[str]]
+        A list of file intervals in sclk string format
+    latest_sclk: Path
+        The latest clock kernel used for the above calculations
+    latest_lsk: Path
+        The latest leapsecond kernel used for the above calculations
+    """
+    # Format the data to insert
+    spice_params = {
+        "ingestion_date": datetime.now(),
+        "kernel_type": spice_object.spice_metadata["type"],
+        "version": spice_object.spice_metadata["version"],
+        "file_name": str(spice_object.filename.name),
+        "file_root": str(spice_object.filename.name).replace(
+            str(spice_object.spice_metadata["version"]), ""
+        ),
+        "min_date_j2000": file_coverage_j2000[0][0],
+        "max_date_j2000": file_coverage_j2000[-1][-1],
+        "file_intervals_j2000": file_coverage_j2000,
+        "min_date_datetime": file_coverage_datetime[0][0],
+        "max_date_datetime": file_coverage_datetime[-1][-1],
+        "file_intervals_datetime": [
+            [dt.isoformat() for dt in sublist] for sublist in file_coverage_datetime
+        ],
+        "min_date_sclk": file_coverage_sclk[0][0],
+        "max_date_sclk": file_coverage_sclk[-1][-1],
+        "file_intervals_sclk": file_coverage_sclk,
+        "lsk_kernel": str(latest_lsk),
+        "sclk_kernel": str(latest_sclk),
+    }
+
+    with db.Session() as session:
+        # Execute the statement as a single "insert-or-update" operation
+        stmt = (
+            insert(models.SPICEFiles)
+            .values(**spice_params)
+            .on_conflict_do_update(
+                index_elements=["file_name"],  # or name of a unique constraint
+                set_={  # Remove the "file_name" from the update dict
```

Comment:
Since it is unique and you've "found" this entry, does it matter if we update it or not here? i.e. could you just `set_=spice_params`

Reply:
Yeah I tried that initially and it still gave an error unfortunately, it still doesn't like you trying to modify the file_name field (even though it is the same)